File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse2003WinePairing.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse2003WinePairingProductMatches
import scala.collection.immutable.Seq

/**
 * 
 * @param pairedWines 
 * @param pairingText 
 * @param productMatches 
 */
case class InlineResponse2003WinePairing(pairedWines: Seq[String],
                pairingText: String,
                productMatches: Seq[InlineResponse2003WinePairingProductMatches]
                )

object InlineResponse2003WinePairing {
    /**
     * Creates the codec for converting InlineResponse2003WinePairing from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse2003WinePairing] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse2003WinePairing] = deriveEncoder
}

File: dart/com/spoonacular/client/model/inline_response2003_wine_pairing.dart
part of openapi.api;

class InlineResponse2003WinePairing {
  
  List<String> pairedWines = [];
  
  String pairingText = null;
  
  List<InlineResponse2003WinePairingProductMatches> productMatches = [];
  InlineResponse2003WinePairing();

  @override
  String toString() {
    return 'InlineResponse2003WinePairing[pairedWines=$pairedWines, pairingText=$pairingText, productMatches=$productMatches, ]';
  }

  InlineResponse2003WinePairing.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['pairedWines'] == null) {
      pairedWines = null;
    } else {
      pairedWines = (json['pairedWines'] as List).cast<String>();
    }
    if (json['pairingText'] == null) {
      pairingText = null;
    } else {
          pairingText = json['pairingText'];
    }
    if (json['productMatches'] == null) {
      productMatches = null;
    } else {
      productMatches = InlineResponse2003WinePairingProductMatches.listFromJson(json['productMatches']);
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (pairedWines != null)
      json['pairedWines'] = pairedWines;
    if (pairingText != null)
      json['pairingText'] = pairingText;
    if (productMatches != null)
      json['productMatches'] = productMatches;
    return json;
  }

  static List<InlineResponse2003WinePairing> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse2003WinePairing>() : json.map((value) => new InlineResponse2003WinePairing.fromJson(value)).toList();
  }

  static Map<String, InlineResponse2003WinePairing> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse2003WinePairing>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse2003WinePairing.fromJson(value));
    }
    return map;
  }
}

